builder/vsphere/iso/step_cleanup.go

```diff
@@ -0,0 +1,36 @@
+//go:generate packer-sdc struct-markdown
+//go:generate packer-sdc mapstructure-to-hcl2 -type CleanupConfig
+
+package iso
+
+import (
+	"context"
+
+	"github.com/hashicorp/packer-plugin-sdk/multistep"
+)
+
+type CleanupConfig struct {
+	// If set to true, the VM will be destroyed after the builder completes
+	Destroy bool `mapstructure:"destroy"`
```

Comment:
To help piggy back off the clean up logic already in place for the StepCreate we can actually add this configuration option into the [CreateConfig Configuration](https://github.com/hashicorp/packer-plugin-vsphere/blob/9fc0b63ff62b9460ea15961a04d0fc645e9b520d/builder/vsphere/iso/step_create.go#L61) structure and get rid of this new configuration type.

Reply:
[827d6ba](https://github.com/hashicorp/packer-plugin-vsphere/pull/204/commits/827d6baf75f7f9e582a1cab57a43e978d8743edd) and [3495c47](https://github.com/hashicorp/packer-plugin-vsphere/pull/204/commits/3495c4718c1e7b4815133769a45b95b93076de76) should move this feature to the `step_create` as suggested. Yet, if you think we can address #168 as part of this PR, it might be interesting to finally create a common `step_cleanup` used in both builders. WDYT?